model/project_ref.go

```diff
@@ -1339,6 +1339,32 @@ func FindFirstProjectRef() (*ProjectRef, error) {
 	return &projectRef, nil
 }
 
+// FindAllMergedTrackedProjectRefsWithRestrictedProjects returns all project refs in the db
+// that are currently being tracked (i.e. their project files
+// still exist and the project is not hidden) and are not restricted projects.
+// You can optionally pass in a list of restricted projects the user has permission to view to include in the results.
+func FindAllMergedTrackedProjectRefsWithRestrictedProjects(restrictedProjectRefIds []string) ([]ProjectRef, error) {
```

Comment:
I think this function comes from a good place of doing everything in the query but is a bit over engineered. Can't we just query the list of IDs? We already know the list of projectIds we want to return.

Also groupProjects already handles hidden. It feels to me like groupProjects could have an additional parameter to optionally filter out disabled projects, and then the logic from https://github.com/evergreen-ci/evergreen/blob/dc332be736935205d2e498e1000952f2f3e368a4/graphql/query_resolver.go#L533 can be pretty much reused. 

Some separate things are that the title and the restrictedProejctRefIds field seem misnamed (we're actually passing in all projectIds the user can see, and we're not returning _all_ merged tracked projects).